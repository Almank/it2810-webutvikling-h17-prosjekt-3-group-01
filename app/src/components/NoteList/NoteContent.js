/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import { Text, View, StyleSheet, TextInput } from 'react-native';
import { Button } from 'react-native-elements';

export class NoteContent extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            textfield: "",
        };
        this.handleChange = this.handleChange.bind(this);
        this.returnToLastView = this.returnToLastView.bind(this);
        this.handleDelete = this.handleDelete.bind(this);
    }

    returnToLastView(){
        this.props.navigation.goBack();
    }

    handleChange(text){
        let data = this.state;
        data.textfield = text;
        this.setState(data);
    }

    handleDelete(){
        this.props.navigation.state.params.onPress();
        this.returnToLastView();
    }

    render(){
        return(
            <View>
                <View style={styles.header}>
                    <Button buttonStyle={styles.headerButton}
                            title="< Back"
                            fontWeight={"bold"}
                            onPress={this.returnToLastView}/>
                </View>
                <View style={styles.noteContainer}>
                    <View style={styles.hr}/>
                    <Text style={styles.headerTitle}>{this.props.navigation.state.params.title}</Text>
                    <View style={styles.hr}/>
                    <TextInput style={styles.noteField}
                               onChangeText={(text) => this.handleChange(text)}
                               value={this.state.text}
                               multiline={true}
                               numberOfLines={4}
                    />
                    <Button title="Delete Note"
                            fontWeight={'bold'}
                            fontSize={15}
                            onPress={this.handleDelete}
                            buttonStyle={styles.DeleteButton}
                    />
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    header: {
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        backgroundColor: 'white',
    },
    headerButton: {
        marginTop: 5,
        marginBottom: 5,
        marginLeft: -10,
        backgroundColor: '#35393a',
    },
    headerTitle: {
        textAlign: 'center',
        marginTop: 5,
        marginBottom: 5,
        fontSize: 15,
        fontWeight: 'bold',
    },
    hr: {
        borderBottomColor: 'grey',
        borderBottomWidth: 1,
    },
    noteContainer: {
        display: 'flex',
    },
    noteField: {
        backgroundColor: 'white',
        height: 300,
    },
    DeleteButton: {
        marginTop: 10,
        backgroundColor: '#8f0500',
    }
});