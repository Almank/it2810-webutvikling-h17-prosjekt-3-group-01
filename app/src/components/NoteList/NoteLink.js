import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet, TouchableHighlight } from 'react-native';

export class NoteLink extends React.Component {
    constructor(props){
        super(props);
        this.navigateToContent = this.navigateToContent.bind(this);
        this.onDelete = this.onDelete.bind(this);
        this.onChildContentChange = this.onChildContentChange.bind(this);
    }

    navigateToContent(){
        this.props.navigation.navigate('Content', {
            title: this.props.title,
            content: this.props.content,
            navigation: this.props.navigation,
            onPress: this.onDelete,
            onContentChange: this.onChildContentChange,
        });
    }

    onChildContentChange(title, text){
        this.props.onContentChange(title, text);
    }

    onDelete(){
        this.props.onClick(this.props.title);
    }

    render(){
        return(
            <TouchableHighlight onPress={this.navigateToContent}>
                <View style={styles.NoteLink}>
                    <Text style={styles.Title}>
                          {this.props.title}
                    </Text>
                    <Text style={styles.Title}>
                        >
                    </Text>
                </View>
            </TouchableHighlight>
        )
    }
}

NoteLink.PropTypes = {
    navigation: PropTypes.navigation,
    title: PropTypes.string,
    content: PropTypes.string,
    index: PropTypes.number,
    key: PropTypes.number.isRequired,
    onContentChange: PropTypes.func,
    onClick: PropTypes.func,
};

const styles = StyleSheet.create({
    NoteLink: {
        display: 'flex',
        paddingTop: 10,
        paddingLeft: 15,
        paddingRight: 15,
        paddingBottom: 10,
        backgroundColor: 'white',
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        borderBottomColor: 'grey',
        borderTopColor: 'grey',
        borderRadius: 1,
        borderWidth: 0.25,
    },
    Title: {
        fontSize: 16,
    },
});