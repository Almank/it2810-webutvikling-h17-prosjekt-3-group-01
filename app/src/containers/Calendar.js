import React from 'react';
import { View, StyleSheet, AsyncStorage, Text, ScrollView } from 'react-native';
import {Week} from '../components/Calendar/Week';
import {AppointmentForm} from "../components/Calendar/AppointmentForm";

export class Calendar extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            dateToday: new Date().toISOString().slice(0, 10),
            dateVal: '',
            timeVal: '',
            titleVal: '',
            textVal: '',
            children: [
                {date: '2017-10-15', time: '14:15', title: 'Hello World', text: 'This is text'},
            ],
        };

        //Binding functions
        this.createAppointment = this.createAppointment.bind(this);
        this.changeContent = this.changeContent.bind(this);

        this.loadData();
    }

    emptyScheduleCheck(){

        //Copy of the state children array.
        let myData =[].concat(this.state.children)

        //Sorting elements based on time, earliest first.
            .sort((a, b) => a.time > b.time)

            //Filtering out appointments on current day.
            .filter(child => child.date === this.state.dateToday)

            //Mapping items from array giving the html the correct values.
            .map((item,i) =>
                <View style={styles.showAppointments} key={i}>
                    <Text style={styles.appointmentItem}>{item.time}</Text>
                    <Text style={styles.appointmentItem}>{item.title}</Text>
                    <Text style={styles.appointmentItem}>{item.text}</Text>
                </View>
            );

        //If there are no plans that day.
        if (myData.length === 0){
            myData = [  <View style={styles.showAppointments} key={0}>
                <Text style={styles.appointmentNoItems}>You don't have any plans!</Text>
            </View>];
        }

        return (
            <ScrollView style={styles.scrollView}>
                {myData}
            </ScrollView>
        );
    }

    changeContent(e){
        let data = this.state;
        data.dateToday = e.props.dateFull;
        this.setState({ dateToday: e.props.dateFull }, () => {
            this.forceUpdate();
        });
        try{
            AsyncStorage.setItem("calendar", JSON.stringify(data));
        } catch (error){
            console.log(error);
        }
    }

    async loadData(){
        let data;
        try {
            data = await AsyncStorage.getItem('calendar');
            if (data !== null){
                data = JSON.parse(data);
                data.dateToday = new Date().toISOString().slice(0, 10);
                this.setState(data);
            } else{
                this.setState({
                    dateToday: new Date().toISOString().slice(0, 10),
                    children: [],
                    dateValue: '',
                    timeValue: '',
                    titleValue: '',
                    textValue: '',
                });
            }
        } catch (error) {
            console.log(error);
        }
    }

    createAppointment(e) {
        //Fetching values from form.
        let dateValue = e[0];
        let timeValue = e[1];
        let titleValue = e[2];
        let textValue = e[3];

        //Validate fields date, time and title. Last field is optional.
        if (this.validateFormDate(dateValue) && this.validateFormTime(timeValue) && this.validateFormTitle(titleValue)){

            //Creating new array with current state children.
            let newStateArray = this.state.children.slice();

            //Pushing new child to array.
            newStateArray.push({date: dateValue, time: timeValue, title: titleValue, text: textValue});

            //Sorting array with earliest appointments first.
            newStateArray.sort((a, b) => a.time > b.time);

            //Creating new updated state.
            let data = {
                children: newStateArray,
                dateToday: new Date().toISOString().slice(0, 10),
            };

            //Setting state.
            this.setState(data);

            //Setting updated values to storage.
            try{
                AsyncStorage.setItem("calendar", JSON.stringify(data));
            } catch (error){
                console.log(error);
            }

        } else {
            //If there are invalid values.
            alert('Error! Do not submit invalid values.')
        }
    }

    validateFormDate(date){
        return (date.length === 10)
    }

    validateFormTime(time){
        return (time.length === 5)
    }

    validateFormTitle(title){
        return (title.length > 0)
    }

    render(){
        return(
            <View style={styles.container}>
                <Week change={this.changeContent} />
                <View style={styles.bottomContainer}>
                    <AppointmentForm getValues={ arr => this.createAppointment(arr) } />
                    {this.emptyScheduleCheck()}
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        marginTop: 40,
        display: 'flex',
        flexDirection: 'column',
        flex: 1,
        marginBottom: 20,
    },
    bottomContainer: {
        display: 'flex',
        flexDirection: 'column',
        flex: 1,
    },
    showAppointments: {
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'center',
    },
    appointmentItem: {
        flexBasis:'33%',
        textAlign:'center',
        fontSize:16,
        flexWrap: 'wrap',
    },
    appointmentNoItems: {
        marginTop: '10%',
        flexBasis:'90%',
        textAlign:'center',
        fontSize:20,
    },
    scrollView: {
        height: '100%',
    }


});



