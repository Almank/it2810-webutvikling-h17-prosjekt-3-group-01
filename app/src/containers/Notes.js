import React from 'react';
import { View, Text } from 'react-native';

export class Notes extends React.Component {
    render(){
        return(
            <Text>Notes</Text>
        )
    }
}