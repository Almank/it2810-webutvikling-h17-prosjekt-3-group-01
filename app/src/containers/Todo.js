import React from 'react';
import {TodoList} from "../components/TodoList/TodoList";

export class Todo extends React.Component {
    render(){
        return(
            <TodoList />
        )
    }
}