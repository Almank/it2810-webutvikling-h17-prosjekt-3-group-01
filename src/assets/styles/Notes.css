.NoteBook{
    display: flex;
    flex-direction: row;
    height: 100%;
}

/* Main flexboxes */
.NoteList{
    width: 20%;
    min-width: 250px;
    height: 100%;
    background-color: transparent;
    overflow: auto;
}

.NoteEdit{
    width: 80%;
    min-width: 430px;
    height: 100%;
}

.NoteEdit .NoteEditContent{
    height: 100%;
}

/* NoteLink Component */
.NoteListTitleContainer {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 10px;
    height: 79px;
}

.NoteListTitleContainer h1 {
    margin: 0;
}

.addNewNoteButton {
    cursor: pointer;
    font-size: 26px;
    color: #35393a;
    padding: 5px;
}

.addNewNoteButton:hover {
    color: #FF9505;
}

.NoteLinkContainer{
    display: flex;
    justify-content: space-between;
    min-height: 60px;
    align-items: center;
    background-color: #F6F6F6;
    border-bottom: 1px solid #35393a;
    cursor: pointer;
}

.NoteLinkContainer:first-child{
    border-top: 1px solid #35393a;
}

.NoteLinkContainer:nth-child(odd){
    background-color: white;
}

.NoteLinkContainer:hover:nth-child(odd){
    background-color: #F6F6F6;
}

.NoteLinkContainer:hover:nth-child(even){
    background-color: white;
}

.NoteLinkContainer a {
    text-decoration: none;
}

.NoteButton{
    display: flex;
    padding-left: 10px;
    text-decoration: none;
    color: #35393a;
    font-size: 10px;
    width: 70%;
    font-family: Geosans, serif;
    align-content: center;
}

.NoteLinkContainer:hover .NoteButton{
    text-decoration: none;
    color: #FF9505;
}

.NoteButton h1 {
    margin: 0;
    font-size: 24px;
}

.RemoveButton{
    border:none;
    background-color: transparent;
    padding: 15px;
    padding-right: 20px;
    z-index: 2;
}

.RemoveButton span {
    z-index: -1;
    color:#35393a;
    font-size: 20px;
}

.RemoveButton:hover span {
    color: #d9534f;
}

/* NoteList - Form component */

.NoteForm{
    display: none;
    justify-content: center;
    padding-top: 10px;
    padding-bottom: 10px;
    padding-left: 10px;
}

.NoteAdd{
    margin-right: 5px;
    background-color: #35393a;
    color: white;
    border: none;
    border-radius: 0 4px 4px 0;
    padding-left: 17.5px;
    padding-right: 17.5px;
}

.NoteAdd:hover {
    background-color: #FF9505;
}

.NoteField{
    flex:1;
    padding: 2.5px 5px 2.5px 5px;
    width: 165px;
    height: 30px;
    border: 1px solid lightgrey;
    border-radius: 4px 0 0 4px;
}

/* NoteContent Component */
.NoteContentContainer{
    padding: 20px;
    height: 95%;
}

.NoteContentContainer h1{
    border-bottom: 1px solid;
    margin-bottom: 10px;
}

.NoteContentContainer textarea{
    max-width: 100%;
    width: 96%;
    padding: 2%;
    height: 85%;
    margin: 0;
    outline: none;
    resize: none;
    border: 1px solid lightgrey;
}

/* Mobile - portrait */
@media only screen and (max-width : 420px) {
    .NoteBook {
        flex-direction: column;

    }
    .NoteList {
        width: 100vw;
        min-width: 0;
        height: auto;
    }
    .NoteEdit {
        width: 100vw;
        min-width: 0;
        height: auto;
    }
    .NoteContentContainer textarea {
        min-height: 30vh;
    }

}