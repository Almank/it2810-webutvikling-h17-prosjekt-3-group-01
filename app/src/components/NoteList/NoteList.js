import React from 'react';
import { View, StyleSheet, TextInput, FlatList, Keyboard, AsyncStorage} from 'react-native';
import { Button, FormLabel } from 'react-native-elements';
import { NoteLink } from "./NoteLink";

export class NoteList extends React.ComponentÂ {
    constructor(props) {
        super(props);
        this.state = {
            notes: {}
        };

        this.onSubmit = this.onSubmit.bind(this);
        this.handleRemoveClick = this.handleRemoveClick.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.handleNoteContentChange = this.handleNoteContentChange.bind(this);
        this.loadData();
    }

    //Inital async load of data that has to be done outside the constructor
    async loadData(){
        let notes;
        try {
            notes = await AsyncStorage.getItem('notes');
            if (notes !== null){
                notes = JSON.parse(notes);
                this.setState({notes: notes['notes']});
            }
        } catch (error) {
            console.log(error);
        }
    }

    updateAsyncStorage(){
        let data = this.state;
        try{
            AsyncStorage.setItem("notes", JSON.stringify(data));
        } catch (error){
            console.log(error);
        }
    }

    //Function for adding new note and updating storage to contain this.
    onSubmit(){
        let data = this.state;
        if(this.state.textfield) {
            data.notes[this.state.textfield] = {
                title: this.state.textfield,
                content: 'Empty note'
            };
            data.textfield = "";
            this.setState(data);
            Keyboard.dismiss();
            this.updateAsyncStorage();
        }
    }

    //Function for handling the removeclick and has to force update due to the state of its parent.
    handleRemoveClick(title){
        let data = this.state;
        delete data.notes[title];
        this.setState(data);
        this.updateAsyncStorage();
    }

    handleChange(text){
        let data = this.state;
        data.textfield = text;
        this.setState(data);
        this.updateAsyncStorage();
    }

    //function for handling changes in content and save to state
    handleNoteContentChange(title, content){
        let data = this.state.notes;
        data[title].content = content;
        this.setState(data);
        this.updateAsyncStorage();
    }

    //Function for rendering all the listed notes.
    renderNote(item, index){
        return(<NoteLink title={item.title}
                         content={item.content}
                         index={index}
                         key={index}
                         navigation={this.props.navigation}
                         onClick={this.handleRemoveClick}
                         onContentChange={this.handleNoteContentChange}
        />);
    }

    //Processes data from state to be used in FlatList
    renderData(){
        let unprocessedData = this.state.notes;
        let data = [];
        for(let key in unprocessedData){
            data.push(unprocessedData[key])
        }
        return data;
    }

    render(){
        return(
            <View style={styles.NoteList}>
                <View style={styles.InputFields}>
                    <TextInput style={styles.TitleInput}
                               onChangeText={(text) => this.handleChange(text)}
                               value={this.state.text}
                               placeholder="Enter title here..."
                    />
                    <Button raised
                            backgroundColor="#FF9505"
                            underlayColor="#FF9505"
                            color="white"
                            title="ADD NOTE"
                            borderColor="rgba(155,155,155,0.5)"
                            style={styles.AddInput}
                            fontSize={16}
                            onPress={this.onSubmit}/>
                </View>
                <FlatList
                    key={"flatList"}
                    data={this.renderData()}
                    renderItem={({item, index}) => this.renderNote(item, index)}
                    keyExtractor={(key, index) => index}
                />
            </View>
        )
    }
}

const styles = StyleSheet.create({
    NoteList: {
        display: 'flex',
        backgroundColor: 'white',
        flex: 1,
    },
    InputFields: {
        display: 'flex',
        justifyContent: 'flex-end',
        backgroundColor: '#f2f2f2',
        borderBottomWidth: 2,
        borderColor: '#f2f2f2',
        height: '30%',
        paddingTop: 20,
    },
    TitleInput: {
        marginLeft: 15,
        marginRight: 15,
        height: '28%',
        borderBottomWidth: 0.99,
        borderColor: 'grey',
        borderRadius: 2,
        borderWidth: 0.95,
        backgroundColor: 'white',
    },
    AddInput: {
        display: 'flex',
        marginTop: 10,
        marginBottom: 10,
        width: '50%',
        marginLeft:'25%',
        marginRight:'25%',
        flexBasis: 50,
        justifyContent: 'center',
        alignItems: 'center',
    },
});