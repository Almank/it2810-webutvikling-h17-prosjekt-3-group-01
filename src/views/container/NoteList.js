/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import '../../assets/styles/Notes.css';
import {NoteLink} from "../components/NoteList/NoteLink";

export class NoteList extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            notes: this.props.notes,
        };

        this.onSubmit = this.onSubmit.bind(this);
        this.handleRemoveClick = this.handleRemoveClick.bind(this);
    }

    onSubmit(event){
        const data = this.state.notes;
        data[event.target.title.value] = {
            title: event.target.title.value,
        };
        this.setState({notes: data});
        this.props.updateStorage();
        this.createNewNote();
        event.preventDefault();
    }

    handleRemoveClick(event){
        let title = event.target.value;
        this.props.removeClick(title);
        this.forceUpdate();
    }

    renderListedNotes(){
        let dataSet = [];
        const data = this.state.notes;
        let index = 0;
        for(let key in data){
            dataSet.push(<NoteLink title={ data[key].title }
                                   content={data[key].content}
                                   key={index}
                                   onClick={this.handleRemoveClick}
            />);
            index++;
        }
        return dataSet;
    }

    createNewNote() {
        let element = document.querySelector('.NoteForm');
        element.style.display = element.style.display === 'flex' ? '' : 'flex';
    }

    render(){
        return (
            <div className="NoteList">
                <div className="NoteListTitleContainer">
                    <h1>Notes</h1>
                    <div className="addNewNoteButton" onClick={this.createNewNote}>
                        <span className={'glyphicon glyphicon-plus'}></span>
                    </div>
                </div>
                <form className="NoteForm" onSubmit={this.onSubmit}>
                    <input className="NoteField" name="title" type="text"/>
                    <input className="NoteAdd" type="submit" value="ADD"/>
                </form>
                <div className="CurrentNotes">
                    { this.renderListedNotes() }
                </div>
            </div>
        );
    }
}
