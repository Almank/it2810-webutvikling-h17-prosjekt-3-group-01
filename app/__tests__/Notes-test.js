/**
 * Sample React Native Snapshot Test
 */
'use strict';

import 'react-native';
import React from 'react';
import {Notes} from '../src/containers/Notes';
import {NoteList} from '../src/components/NoteList/NoteList';

const wrapper = shallow(<Notes />);
const instWrapper = wrapper.instance();

const wrapper_noteList = shallow(<NoteList />);
const instWrapper_noteList = wrapper_noteList.instance();

//testing that the Notes view renderes correctly, aswell as testing functionality regarding adding notes, changing notes, and removing notes
describe('<Notes />', () => {
	//test if the Notes view renders correctly
	it('renders correctly', () => {
	  expect(wrapper).toMatchSnapshot();
	});
	//test if the NoteList component is rendered correctly
	describe('<NoteList />', () => {
		it('should render correctly', () => {
			const wrapper = shallow(<NoteList />);
			expect(wrapper).toMatchSnapshot();
		});
	});
	//test if change is handled correctly, making sure that the change is the title actually used when testing creation of note
	describe('NoteList.handleChange()', () => {
		it('handles change', () => {
			instWrapper_noteList.handleChange("test_data");
			expect(instWrapper_noteList.state.textfield).toBe("test_data");
		});
	});
	//test if a note is created with the title "test_data" and that it contains default content
	describe('NoteList.onSubmit()', () => {
		it('adds test_data note to notes', () => {
			instWrapper_noteList.onSubmit();
			expect(instWrapper_noteList.state.notes['test_data']['content']).toBe('Write your content here');
		});
	});
	//test if the note titled "test_data" is changed correctly, new content is "test content"
	describe('NoteList.handleNoteContentChange()', () => {
		it('changes content of test note', () => {
			instWrapper_noteList.handleNoteContentChange('test_data', 'test content');
			expect(instWrapper_noteList.state.notes['test_data']['content']).toBe('test content');
		});
	});
	//test if the note titled "test_data" is properly removed
	describe('NoteList.handleRemoveClick()', () => {
		it('removes test note', () => {
			instWrapper_noteList.handleRemoveClick('test_data');
			expect(instWrapper_noteList.state.notes['test_data']).toBe(undefined);
		});
	});

});


