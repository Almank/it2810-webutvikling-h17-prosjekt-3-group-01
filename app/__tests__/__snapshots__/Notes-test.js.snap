// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notes /> <NoteList /> should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#696969",
      "display": "flex",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "lightgrey",
        "borderBottomWidth": 2,
        "borderColor": "grey",
        "display": "flex",
        "justifyContent": "flex-end",
      }
    }
  >
    <FormLabel
      labelStyle={
        Object {
          "fontSize": 20,
        }
      }
    >
      Title
    </FormLabel>
    <TextInput
      onChangeText={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "marginLeft": 15,
          "marginRight": 15,
        }
      }
    />
    <Button
      backgroundColor="green"
      color="white"
      fontSize={20}
      fontWeight="bold"
      onPress={[Function]}
      raised={true}
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
        }
      }
      title="Create New Note"
      underlayColor="lightgreen"
    />
  </View>
  <FlatList
    data={Array []}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    key="flatList"
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;

exports[`<Notes /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": -65,
    }
  }
>
  <NavigationContainer />
</View>
`;
